glue/viewers/image/state.py

```diff
@@ -543,8 +543,15 @@ def __init__(self, layer=None, viewer_state=None, **kwargs):
 
         self.update_from_dict(kwargs)
 
+        if isinstance(self.layer.style.preferred_cmap, str):
+
+            for i, element in enumerate(colormaps.members):
+                if element[0] == self.layer.style.preferred_cmap:
+                    self.layer.style.preferred_cmap = element[1]
```

Comment:
Maybe you could actually code this up in the property on the Visual class in case we end up using this elsewhere too?

Reply:
Sure